spec/02-integration/000-new-dao/02-db_core_entities_spec.lua

```diff
@@ -410,10 +430,15 @@ for _, strategy in helpers.each_strategy() do
             code        = Errors.codes.SCHEMA_VIOLATION,
             name = "schema violation",
             strategy    = strategy,
-            message     = "schema violation (paths: field required for entity check)",
-            fields      = {
-              paths = "field required for entity check",
-            }
+            message  = unindent([[
+              schema violation
+              (when updating, at least one of these fields must be non-empty: 'methods', 'hosts', 'paths')
+            ]], true, true),
+            fields   = {
+              ["@entity"] = {
+                "when updating, at least one of these fields must be non-empty: 'methods', 'hosts', 'paths'",
+              }
+            },
```

Comment:
Moving the previous conversation here, but not considering it a blocker either (bigger fish to fry):

What I was saying is that it seems that both this test and the one at L.350 look alike:

- They both create a Service with `n` amount of semi-required fields
- They both issue an update which sets those `n` semi-required fields to `null`

This _seems_ to be a case of "1 feature breaks and now N tests are failing" because they were testing the same thing.

Maybe you want to test both one and many semi-optional fields being unset, in which case maybe those names can help:
- `fails trying to unset an interdependent field by itself`
- `fails trying to unset several interdependent fields at once`

Or I am missing something else that differentiates those tests.

Again, only clarifying my previous point, I think the main concern I had with this PR personally was the lack of "when updating ..." context in the error message, which is now addressed.

Reply:
Oh, I see.

These two tests as different because they document different aspects of system behavior:

* the test at 350 starts with `hosts[]=example.com methods[]=GET` and tries to update with `methods=null`. A user could reasonably expect this to work, but it doesn't because we don't do read-before-write. This test documents this.

* the test at 418 starts with `hosts[]=example.com methods[]=GET` and tries to update with `hosts=null methods=null`. A user could **not** reasonably expect this to work because it would break the constraint. This test verifies that indeed it does not work.

So, the second test fails for the same reason of the first test, but in the future the system behavior could be changed so that the first test becomes o a success-test but the second will still need to be a fail-test.

Reply:
Ok, that is what I thought (that they currently test the same thing), lgtm if intentional then. I think I will just move the tests closer to each other and name them similarly then.